src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantDetector.java

```diff
@@ -879,6 +852,153 @@ private static boolean alignmentIsDisjointFromAlphaOmega(final SimpleInterval al
+            int result = affectedRefRegion.hashCode();
+            result = 31 * result + Arrays.hashCode(altSeq);
+            result = 31 * result + segmentsAndAltArrange.hashCode();
+            return result;
+        }
+
+        private Map<String, String> typeSpecificExtraAttributes() {
+            final Map<String, String> attributes = new HashMap<>();
+            attributes.put(CPX_EVENT_ALT_ARRANGEMENTS,
+                    segmentsAndAltArrange.getAltArrangementString());
+            if ( ! segmentsAndAltArrange.referenceSegments.isEmpty() ) {
+                attributes.put(CPX_SV_REF_SEGMENTS,
+                        String.join(VCFConstants.INFO_FIELD_ARRAY_SEPARATOR,
+                                segmentsAndAltArrange.referenceSegments.stream()
+                                        .map(SimpleInterval::toString).collect(Collectors.toList())));
+            }
+            return attributes;
+        }
+
+        VariantContext toVariantContext(final ReferenceMultiSource reference) throws IOException {
+
+            final CpxVariantType cpxVariant = new CpxVariantType(affectedRefRegion, typeSpecificExtraAttributes());
+            final SimpleInterval pos = new SimpleInterval(affectedRefRegion.getContig(), affectedRefRegion.getStart(), affectedRefRegion.getStart());
+
+            final VariantContextBuilder vcBuilder = new VariantContextBuilder()
+                    .chr(affectedRefRegion.getContig()).start(affectedRefRegion.getStart()).stop(affectedRefRegion.getEnd())
+                    .alleles(AnnotatedVariantProducer.produceAlleles(pos, reference, cpxVariant))
+                    .id(cpxVariant.getInternalVariantId())
+                    .attribute(SVTYPE, cpxVariant.toString())
+                    .attribute(VCFConstants.END_KEY, affectedRefRegion.getEnd())
+                    .attribute(SVLEN, cpxVariant.getSVLength())
+                    .attribute(SEQ_ALT_HAPLOTYPE, new String(altSeq));
+
+            cpxVariant.getTypeSpecificAttributes().forEach(vcBuilder::attribute);
+            return vcBuilder.make();
+        }
+    }
+
+    // evidence used for producing the novel adjacency
+    private static Map<String, String> standardAttributes(final Iterable<AnnotatedContig> annotatedContigs) {
+        final List<AnnotatedContig> sorted = Utils.stream(annotatedContigs)
+                .sorted(Comparator.comparing(tig -> tig.tigWithInsMappings.getSourceContig().contigName))
+                .collect(Collectors.toList());
+
+        final Map<String, String> attributeMap = new HashMap<>();
+        attributeMap.put(GATKSVVCFConstants.TOTAL_MAPPINGS,    String.valueOf(sorted.size()));
+        attributeMap.put(GATKSVVCFConstants.CONTIG_NAMES,      sorted.stream().map(annotatedContig -> annotatedContig.tigWithInsMappings.getSourceContig().contigName).collect(Collectors.joining(VCFConstants.INFO_FIELD_ARRAY_SEPARATOR)));
+
+        // TODO: 2/1/18 question: show we introduce new annotations for them?
```

Comment:
What do you mean by "introduce new annotations"? I'd add these in so that your calls are equivalent to the calls from the current tool. However, to make the annotations for these complex calls equivalent to those for deletions, etc, I'd look only at the flanking alignments that anchor the variant to the reference -- ie. the head and tail alignments from the contig. Does that make sense?